app/views/contexts/preview/preview_question.py

```diff
@@ -0,0 +1,131 @@
+from re import findall
+from typing import Any, Iterator, Optional, Union
+
+from werkzeug.datastructures import ImmutableDict
+
+from app.questionnaire import QuestionnaireSchema, QuestionSchemaType
+
+
+class PreviewQuestion:
+    def __init__(
+        self, question_schema: QuestionSchemaType, survey_data: ImmutableDict[str, str]
+    ):
+        self.title = question_schema.get("title")
+        self.answers = self._build_answers(question_schema, survey_data)
+        self.descriptions = self._build_descriptions(question_schema, survey_data)
+        self.guidance = self._build_question_guidance(question_schema)
+        self.text_length = self._get_length(question_schema.get("answers", None))
+        self.instruction = question_schema.get("instruction") or None
+        self.answer_description = self._build_answer_descriptions(
+            iter(question_schema["answers"])
+        )
+        self.answer_guidance = self._build_answer_guidance(
+            answers=iter(question_schema["answers"])
+        )
+        self.survey_data = survey_data
+
+    def _build_answers(
+        self, question_schema: QuestionSchemaType, survey_data: ImmutableDict[str, str]
+    ) -> list[Optional[str]]:
+        answers = []
+        for answer in iter(question_schema["answers"]):
```

Comment:
There are a few uses of `iter`. Why are they needed?. They are all already iterable.

Reply:
Removed all the `iter` uses after refactor.